fsx/main.tf

```diff
@@ -0,0 +1,327 @@
+  type                            = "MANAGED"
+  service_role                    = aws_iam_role.batch.arn
+
+  compute_resources {
+    type               = "EC2"
+    instance_type      = ["m5.large"]
+    max_vcpus          = var.workflow_max_vcpus
+    subnets            = [aws_subnet.public.id]
+    security_group_ids = [aws_security_group.all.id]
+    instance_role      = aws_iam_instance_profile.workflow.arn
+
+    launch_template {
+      launch_template_id = aws_launch_template.workflow.id
+      version            = aws_launch_template.workflow.latest_version
+    }
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_batch_job_queue" "workflow" {
+  name                 = "${var.environment_tag}-workflow"
+  state                = "ENABLED"
+  priority             = 1
+  compute_environments = [aws_batch_compute_environment.workflow.arn]
+
+  # miniwdl-aws-submit only needs to be given the workflow queue name because it detects other
+  # infrastructure defaults from these tags.
+  tags = {
+    DefaultTaskQueue = aws_batch_job_queue.task.name
+  }
+}
+
+/**************************************************************************************************
+ * IAM roles
+ *************************************************************************************************/
+
+# For Batch EC2 worker instances running WDL tasks
+resource "aws_iam_role" "task" {
+  name = "${var.environment_tag}-task"
+
+  assume_role_policy = <<-EOF
+  {"Statement":[{"Principal":{"Service":"ec2.amazonaws.com"},"Sid":"","Effect":"Allow","Action":"sts:AssumeRole"}],"Version":"2012-10-17"}
+  EOF
+
+  # The following managed policies are convenient to keep this concise, but they're more powerful
+  # than strictly needed: scopes can all be restricted to specific resources rather than
+  # account-wide.
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+    "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess",
+  ]
+}
+
+# For Batch EC2 worker instances running miniwdl itself
+resource "aws_iam_role" "workflow" {
+  name = "${var.environment_tag}-workflow"
+
+  assume_role_policy = <<-EOF
+  {"Statement":[{"Principal":{"Service":"ec2.amazonaws.com"},"Sid":"","Effect":"Allow","Action":"sts:AssumeRole"}],"Version":"2012-10-17"}
+  EOF
+
+  # The following managed policies are convenient to keep this concise, but they're more powerful
+  # than strictly needed. The scopes can all be restricted to specific resources rather than
+  # account-wide; and while certain "write" permissions to Batch are needed, it's less than
+  # "FullAccess".
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+    "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess",
+    "arn:aws:iam::aws:policy/AWSBatchFullAccess",
```

Comment:
It might be possible to limit the Batch access to specific resources such as the work queue and cluster?

Reply:
Yea, the comment above says as much. Just trying to keep the config succinct.